File: ios/ShareStory.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(ShareStory, NSObject)

- (dispatch_queue_t)methodQueue
{
    return dispatch_get_main_queue();
}

+ (BOOL)requiresMainQueueSetup
{
    return YES;
}

RCT_EXTERN_METHOD(isInstagramAvailable: (RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(shareInstagramStory:(NSDictionary *)config
                  resolver: (RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject)

@end

File: ios/ShareStory.swift
@objc(ShareStory)
class ShareStory: NSObject {
    let instagramScheme = URL(string: "instagram-stories://share?source_application")

    @objc
    func isInstagramAvailable(_ resolve: RCTPromiseResolveBlock,
                              rejecter reject: RCTPromiseRejectBlock) -> Void {
        resolve(UIApplication.shared.canOpenURL(instagramScheme!))
    }

    @objc
    func shareInstagramStory(_ config: NSDictionary,
               resolver resolve: RCTPromiseResolveBlock,
               rejecter reject: RCTPromiseRejectBlock) -> Void {
        if(UIApplication.shared.canOpenURL(instagramScheme!) && config["imageUrl"] != nil && config["appId"] != nil){

            let appID = config["appId"]
            let instagramShareScheme = URL(string: "instagram-stories://share?source_application=\(appID)")

            let url = URL(string: config["imageUrl"] as! String)
            var pasteboardItems: Dictionary<String, Any> = [:]

            let data = try? Data(contentsOf: url!) //make sure your image in this url does exist, otherwise unwrap in a if let check / try-catch
            DispatchQueue.main.async {
                pasteboardItems["com.instagram.sharedSticker.backgroundImage"] = data!
                UIPasteboard.general.items = [pasteboardItems]
                UIApplication.shared.openURL(instagramShareScheme!)
            }

            resolve(true)
        }
    }
}

